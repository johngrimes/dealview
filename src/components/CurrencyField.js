import React from 'react'

import InputField from './InputField.js'
import {
  parseNumber,
  formatDollars,
  unformatDollars,
} from '../data/commonTypes.js'

class CurrencyField extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      value: parseNumber(props.value),
    }

    this.handleChange = this.handleChange.bind(this)
  }

  handleChange(value) {
    const raw = unformatDollars(value)
    this.setState(
      () => ({ value: raw }),
      () => {
        if (this.props.onChange) this.props.onChange(raw)
      }
    )
  }

  componentWillReceiveProps(nextProps) {
    this.setState(() => ({ value: parseNumber(nextProps.value) }))
  }

  render() {
    const value = formatDollars(this.state.value)
    const props = {
      ...this.props,
      value,
      type: 'text',
      onChange: this.handleChange,
    }

    return <InputField {...props} />
  }
}

export default CurrencyField
