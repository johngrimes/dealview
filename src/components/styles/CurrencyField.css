.form input.currency[type="text"] {
  width: 170px;
}
