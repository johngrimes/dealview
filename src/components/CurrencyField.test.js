import React from 'react'
import { shallow } from 'enzyme'

import CurrencyField from './CurrencyField.js'

describe('CurrencyField', () => {
  const minProps = {
    name: 'someInput',
  }

  it('should render', () => {
    const props = {
      ...minProps,
      value: '123456',
    }
    const wrapper = shallow(<CurrencyField {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
