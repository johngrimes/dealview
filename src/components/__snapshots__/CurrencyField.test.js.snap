// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyField should render 1`] = `
<InputField
  name="someInput"
  onChange={[Function]}
  type="text"
  value="$123,456"
/>
`;
